File: src/main/java/com/example/climbingapi/mapper/RouteMapper.java
package com.example.climbingapi.mapper;

import com.example.climbingapi.dto.RouteResponse;
import com.example.climbingapi.model.Route;
import org.springframework.stereotype.Component;

@Component
public class RouteMapper {

    public RouteResponse toResponse(Route route) {
        return new RouteResponse(
                route.getId(),
                route.getWallId(),
                route.getName(),
                route.getGrade(),
                route.getLength(),
                route.getStyle(),
                route.getBolts(),
                route.getRopeLengths(),
                route.getFirstAscendant(),
                route.getDescription(),
                route.getCreatedAt()
        );
    }
}

File: src/main/kotlin/com/example/climbingapi/mapper/RouteMapper.kt
package com.example.climbingapi.mapper

import com.example.climbingapi.dto.RouteResponse
import com.example.climbingapi.model.Route
import org.springframework.stereotype.Component

@Component
class RouteMapper {

    fun toResponse(route: Route): RouteResponse {
        return RouteResponse(
            id = route.id,
            wallId = route.wallId,
            name = route.name,
            grade = route.grade,
            length = route.length,
            style = route.style,
            bolts = route.bolts,
            ropeLengths = route.ropeLengths,
            firstAscendant = route.firstAscendant,
            description = route.description,
            createdAt = route.createdAt
        )
    }
}
